### PR TITLE
Strip standard ports off remote target url

### PR DESCRIPTION
**Pull request by poornas**
## Motivation and Context
Url displayed in `mc admin bucket remote ls` shows standard ports in url.

## How to test this PR?
add play as a remote target using `mc admin bucket remote add` . The `mc admin bucket remote ls` should list target url without port.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed

### CODE REVIEW

File: pkg/madmin/remote-target-commands.go [hunk 1]
```diff
@@ -118,7 +118,14 @@ func (t BucketTarget) URL() string {
 	if t.Secure {
 		scheme = "https"
 	}
-	return fmt.Sprintf("%s://%s", scheme, t.Endpoint)
+	hostName := t.Endpoint
```
Comment (harshavardhana) [inline pkg/madmin/remote-target-commands.go]: A better approach would be 

```

```diff --git a/pkg/madmin/remote-target-commands.go b/pkg/madmin/remote-target-commands.go
index 61993d765..9b9280817 100644
--- a/pkg/madmin/remote-target-commands.go
+++ b/pkg/madmin/remote-target-commands.go
@@ -118,7 +118,15 @@ func (t BucketTarget) URL() string {
        if t.Secure {
                scheme = "https"
        }
-       return fmt.Sprintf("%s://%s", scheme, t.Endpoint)
+       urlStr := fmt.Sprintf("%s://%s", scheme, t.Endpoint)
+       u, err := url.Parse(urlStr)
+       if err != nil {
+               return urlStr
+       }
+       if u.Port() == "80" || u.Port() == "443" {
+               u.Host = u.Hostname()
+       }
+       return u.String()
 }
 
 // Empty returns true if struct is empty.
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10498-abeaaa4

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :x: [more...](#cc2dc014-c23b-47c9-83c7-65caf0edb6ec) |
| mint-zoned.sh | :x: [more...](#7456f819-16ec-44dc-9832-5aa43ae020af) |
<a name="7456f819-16ec-44dc-9832-5aa43ae020af"></a>
### 10498-0e1025a/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30304
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 799b316bb7fb:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 3 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 45 seconds
(4/15) Running aws-sdk-ruby tests ... done in 6 seconds
(5/15) Running awscli tests ... done in 2 minutes and 10 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 1 minutes and 10 seconds
(8/15) Running minio-dotnet tests ... done in 53 seconds
(9/15) Running minio-go tests ... done in 2 minutes and 53 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 41 seconds
(11/15) Running minio-js tests ... FAILED in 28 seconds
{
  "name": "minio-js",
  "function": "fPutObject(bucketName, objectName, filePath, metaData)",
  "args": "bucketName:minio-js-test-4319b447-34fd-4fff-8a2a-fb1ff38c8285, objectName:datafile-65-MB, filePath:/tmp/datafile-65-MB",
  "duration": 925,
  "status": "FAIL",
  "error": "Error: read ECONNRESET at TCP.onStreamRead (internal/stream_base_commons.js:209:20)"
}
(11/15) Running minio-py tests ... done in 3 minutes and 45 seconds
(12/15) Running s3cmd tests ... done in 22 seconds
(13/15) Running s3select tests ... done in 17 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="cc2dc014-c23b-47c9-83c7-65caf0edb6ec"></a>
### 10498-0e1025a/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:31309
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp cc12f480a2b0:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 2 minutes and 9 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 1 minutes and 13 seconds
(8/15) Running minio-dotnet tests ... done in 46 seconds
(9/15) Running minio-go tests ... done in 2 minutes and 23 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 33 seconds
(11/15) Running minio-js tests ... FAILED in 3 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 15,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:678:10) at endReadableNT (_stream_readable.js:1244:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(11/15) Running minio-py tests ... done in 2 minutes and 53 seconds
(12/15) Running s3cmd tests ... done in 21 seconds
(13/15) Running s3select tests ... done in 11 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10498-0e1025a

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10498-cc533cd

**Review by kannappanr — APPROVED** [review]
LGTM

**Pull request merged**